ui/app/contexts/metametrics.new.js

```diff
@@ -4,33 +4,49 @@
  * metrics system. This file implements Segment analytics tracking.
  */
 import React, {
-  useRef,
   Component,
   createContext,
   useEffect,
-  useMemo,
+  useRef,
+  useCallback,
 } from 'react'
-import { useSelector } from 'react-redux'
+import { useDispatch, useSelector } from 'react-redux'
 import PropTypes from 'prop-types'
-import { useLocation, matchPath, useRouteMatch } from 'react-router-dom'
+import { matchPath, useLocation, useRouteMatch } from 'react-router-dom'
 import { captureException, captureMessage } from '@sentry/browser'
 
 import { omit } from 'lodash'
-
 import { getEnvironmentType } from '../../../app/scripts/lib/util'
 import { PATH_NAME_MAP } from '../helpers/constants/routes'
-import { getCurrentLocale } from '../ducks/metamask/metamask'
-import {
-  getCurrentChainId,
-  getMetricsNetworkIdentifier,
-  txDataSelector,
-} from '../selectors'
-import {
-  getTrackMetaMetricsEvent,
-  METAMETRICS_ANONYMOUS_ID,
-  segment,
-} from '../../../shared/modules/metametrics'
+import { txDataSelector } from '../selectors'
+
+import { trackMetaMetricsEvent, trackMetaMetricsPage } from '../store/actions'
+
+// type imports
+/**
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventPayload} MetaMetricsEventPayload
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsEventOptions} MetaMetricsEventOptions
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsPageObject} MetaMetricsPageObject
+ * @typedef {import('../../../shared/constants/metametrics').MetaMetricsReferrerObject} MetaMetricsReferrerObject
+ */
+
+// types
+/**
+ * @typedef {Omit<MetaMetricsEventPayload, 'environmentType' | 'context'>} UIMetricsEventPayload
+ */
+/**
+ * @typedef {Omit<MetaMetricsEventOptions, 'metaMetricsSendCount'>} UIMetricsEventOptions
```

Comment:
There is no `metaMetricsSendCount` property in the `MetaMetricsEventOptions` type:
```suggestion
 * @typedef {MetaMetricsEventOptions} UIMetricsEventOptions
```

Reply:
Thanks!